services/src/main/java/org/fao/geonet/api/ApiUtils.java

```diff
@@ -119,13 +119,15 @@ public static String getInternalId(String uuidOrInternalId, Boolean approved)
             id = String.valueOf(metadata.getId());
         }
 
-        if (StringUtils.isEmpty(id)) {
+        if (!StringUtils.hasLength(id) && Lib.type.isInteger(id)) {
             //It wasn't a UUID
             id = String.valueOf(metadataUtils.findOne(uuidOrInternalId).getId());
         } else if (Boolean.TRUE.equals(approved)) {
             //It was a UUID, check if draft or approved version
-            id = String.valueOf(ApplicationContextHolder.get().getBean(MetadataRepository.class)
-                .findOneByUuid(uuidOrInternalId).getId());
+            AbstractMetadata approvedMetadata = ApplicationContextHolder.get().getBean(MetadataRepository.class).findOneByUuid(uuidOrInternalId);
+            if (approvedMetadata != null) {
+                id = String.valueOf(approvedMetadata.getId());
+            }
         }
 
         if (StringUtils.isEmpty(id)) {
```

Comment:
```suggestion
        if (!StringUtils.hasLength(id)) {
```

Replace deprecated method